### DISCUSSION TITLE
Concurrency and long running workflow

### DISCUSSION THREAD

**Discussion by SachinMehndiratta** [elsa-workflows/elsa-core#2643] [Q&A]
I have a few quick questions if someone can help me understand

- Whether my long running workflows in sleep mode consumes the memory or any thread?
- Can I terminate the workflow?
- Is there any control on number of threads I can dedicate for workflow instances? What is the max limit, if any?

Thanks,

**Comment by sfmskywalker** [comment]
> 1. Whether my long running workflows in sleep mode consumes the memory or any thread?

None whatsoever; workflows in sleep mode are persisted in the DB only and do not exist in memory. This allows you to have potentially infinite numbers of workflow instances in your system without all of them being loaded in memory. This of course also depends on how many workflow instances execute at any given time.

> 2. Can I terminate the workflow?

There's currently no simple to use API to stop (cancel) a workflow, but it can be done by updating the workflow instance's `Status` property and setting it to `Cancelled`. But it might be better to create an issue requesting this feature.

> 3. Is there any control on number of threads I can dedicate for workflow instances? What is the max limit, if any?

There's no logical limit that I know of - only a limit in terms of what the OS allows.
